### PR TITLE
EC2 and pricing updates [v2]

### PR DESCRIPTION
**Pull request by pmhahn**
The EC2 Instance Type information contains some errors (especially the disk size for the m3 instances), conversion errors between GiB an M[i]B, disk count, ..
Also g2.8xlarge and t2.large are missing.

### CODE REVIEW

File: contrib/scrape-ec2-prices.py [hunk 1]
```diff
@@ -112,6 +114,18 @@
     'us-gov-west-1': 'ec2_us_govwest'
 }
 
+INSTANCE_SIZES = [
+    'micro',
+    'small',
+    'medium',
+    'large',
+    'xlarge',
+    'x-large'
```
Comment (jimbobhickville) [inline contrib/scrape-ec2-prices.py]: I think you missed a trailing comma here.

**Comment by jimbobhickville** [comment]
Wow, that must have been super tedious.  I'm surprised Amazon doesn't provide an API for this information.  It looked good to me apart from that missing comma, but I don't have a lot of context around AWS so someone else more familiar should probably review.

File: contrib/scrape-ec2-prices.py (hunk 1 above)
Comment (pmhahn) [inline contrib/scrape-ec2-prices.py]: Thanks, added and force-pushed

**Comment by pmhahn** [comment]
I only slightly improved the sorting of EC2 scraper written mainly by Tomaz Muraus, which basically used the data from https://a0.awsstatic.com/pricing/1/ec2/linux-od.min.js. That could also be used to fill in the data in <libcloud/compute/drivers/ec2.py>, but that's for another day.
But yes, finding those mistakes of mixing 1000 and 1024 was some work ;-)

File: libcloud/compute/drivers/ec2.py [hunk 2]
```diff
@@ -1317,87 +1497,6 @@
             'os1.8xlarge'
         ]
     },
-    'eu-west-2': {
```
Comment (tonybaloney) [inline libcloud/compute/drivers/ec2.py:1320]: @pmhahn  Why has outscale been completely removed?

**Comment by tonybaloney** [comment]
@pmhahn that basically undoes https://github.com/apache/libcloud/pull/645 I think you made an error in merging (resyncing from trunk).

**Comment by Kami** [comment]
Yeah, there is `contrib/scrape-ec2-prices.py`, but it's possible that it's not working anymore since some of those .js files have been deprecated for a while now :disappointed:

**Comment by pmhahn** [comment]
@tonybaloney: My reading is, that #645 is wrong, since OUTSCALE_(SAS|INC)_REGION_DETAILS is a dictionary and the keys "eu-west-2", "us-east-2" are used multiple times:

```
$ curl -s https://raw.githubusercontent.com/apache/libcloud/trunk/libcloud/data/pricing.json | sed -rne '/^ {8}"/p' | sort | uniq -d
        "osc_inc_eu_west_2": {
        "osc_inc_us_east_2": {
        "osc_sas_eu_west_2": {
        "osc_sas_us_east_2": {
$ curl -s https://raw.githubusercontent.com/apache/libcloud/trunk/libcloud/compute/drivers/ec2.py | sed -rne "/^OUTSCALE_SAS_REGION_DETAILS/,/^}/{/^ {4}'/p}" | sort | uniq -d
    'eu-west-2': {
    'us-east-2': {
$ curl -s https://raw.githubusercontent.com/apache/libcloud/trunk/libcloud/compute/drivers/ec2.py | sed -rne "/^OUTSCALE_INC_REGION_DETAILS/,/^}/{/^ {4}'/p}" | sort | uniq -d
    'eu-west-2': {
    'us-east-2': {
```

@Kami: You wan't me not not update the EC2 pricing? At least adding the prices for g2.8xlarge and t2.large is required to not break the unit test.

File: libcloud/compute/drivers/ec2.py (hunk 2 above)
Comment (pmhahn) [inline libcloud/compute/drivers/ec2.py:1320]: @tonybaloney No it hasn't been removed conmpletely, only the duplicate entries: it's a JSON _dictionary_, where the key is _unique_ and only the last(?) entry wins:
Warning: the following script will create the files [1-8].py in your current directory!

```
curl -s https://raw.githubusercontent.com/apache/libcloud/trunk/libcloud/compute/drivers/ec2.py | sed -ne '
1567,1593w 1.py
1594,1620w 2.py
1621,1647w 3.py
1648,1674w 4.py

1729,1755w 5.py
1756,1782w 6.py
1783,1809w 7.py
1810,1836w 8.py
'
diff 1.py 2.py ; diff 2.py 3.py ; diff 3.py 4.py
diff 5.py 6.py ; diff 6.py 7.py ; diff 7.py 8.py
```

Comment (jimbobhickville) [inline libcloud/compute/drivers/ec2.py:1320]: Yep, `eu-west-2` exists up above in the elided part of the diff, line 1293.

**Comment by tonybaloney** [comment]
@pmhahn yep good point, i see the mistake now.

**Comment by Kami** [comment]
LGTM

**Comment by tonybaloney** [comment]
Ok @pmhahn I'll close and merge this. FYI @LordShion see above.

**Pull request closed without merging**